app/models/user.rb

```diff
@@ -9,4 +9,40 @@ class User < ApplicationRecord
   has_many :posts
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
+
+  has_many :requested_friendships, foreign_key: 'requester_id', class_name: 'Friendship'
+  has_many :received_friendships, foreign_key: 'receiver_id', class_name: 'Friendship'
+
+  #  has_many :friendships
+  #  has_many :inverse_friendships, :class_name => "Friendship", :foreign_key => "friend_id"
+
+  def friends
+    requested_array = requested_friendships.map { |fr| fr.receiver if fr.status }
+    received_array = received_friendships.map { |fr| fr.requester if fr.status }
+    (requested_array + received_array).compact
+  end
```

Comment:
- [ ] If you implement the double friendship rows properly then this method will become faster to query:
```
def friends
    requested_array = requested_friendships.map { |fr| fr.receiver if fr.status }
    requested_array
  end
```